task-sdk/src/airflow/sdk/bases/notifier.py

```diff
@@ -104,4 +132,23 @@ def __call__(self, *args) -> None:
         try:
             self.notify(context)
         except Exception as e:
-            self.log.exception("Failed to send notification: %s", e)
+            self.log.exception("Failed to send (blocking) notification: %s", e)
+
+    def __await__(self) -> Generator[Any, None, None]:
+        """
+        Make the notifier awaitable.
+
+        Context provided in the constructor is used.
+        """
+        self._update_context(self.context)
+        self.render_template_fields(self.context)
+        try:
+            return self.async_notify(self.context).__await__()
```

Comment:
Two questions here can't we make this a normal async function?

```suggestion
    async def __await__(self) -> Any:
        """
        Make the notifier awaitable.

        Context provided in the constructor is used.
        """
        self._update_context(self.context)
        self.render_template_fields(self.context)
        try:
            return await self.async_notify(self.context)
```

And why do we need the empty generator? I think with the above change that isn't needed anymore?

Reply:
Oh yes. Then how about move all the logic (rendering, error handling) into an async function, and then this fn becomes:

```python
   async def _something(self):
        self._update_context(self.context)
        self.render_template_fields(self.context)
        ... # etc.

    def __await__(self):
        return self._something().__await__()
```
I think that would be cleanrer than heeding to have the empty generator.

Reply:
Or don't support await on the object, and have the api be `await n.async_notify()` directly. Either or.

Reply:
If you've cleaned it up that's call -- my major complaint was the exception handling and having to have the odd-to-me empyt generator